@here/harp-mapview/lib/copyrights/UrlCopyrightProvider.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { AreaCopyrightInfo, CopyrightCoverageProvider } from "./CopyrightProvider";
+
+/**
+ * Copyright provider which retrieves copyright coverage information from provided URL.
+ */
+export class UrlCopyrightProvider extends CopyrightCoverageProvider {
+    private m_cachedCopyrightResponse: Promise<AreaCopyrightInfo[]> | undefined;
+
+    /**
+     * Default constructor.
+     *
+     * @param m_fetchURL URL to fetch copyrights data from.
+     * @param m_baseScheme Scheme to get copyrights from.
+     */
+    constructor(private m_fetchURL: string, private m_baseScheme: string) {
+        super();
+    }
+
+    /** @inheritdoc */
+    getCopyrightCoverageData(): Promise<AreaCopyrightInfo[]> {
+        if (this.m_cachedCopyrightResponse !== undefined) {
+            return this.m_cachedCopyrightResponse;
+        }
+
+        this.m_cachedCopyrightResponse = fetch(this.m_fetchURL)
+            .then(async response => {
+                const json = await response.json();
+                return json[this.m_baseScheme];
+            })
+            .catch(() => []);
```

Comment:
Add proper error handling, in this case at least `logger.error`.

Reply:
Added logging of error.